### PR TITLE
Avoid an exception when response isn't JSON-formatted

### PR DESCRIPTION
**Pull request by pilou-**
This could occur when user provided an incorrect URL.

### CODE REVIEW

File: tests/test_vault_api.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            resp = HTTPResponse(body=io.BytesIO(data), preload_content=False)
+            resp.status = 404
+            resp.reason = 'Not Found'
+            resp.headers = {}
+            return self.build_response(request, resp)
+
+        with unittest.mock.patch('requests.adapters.HTTPAdapter.send', autospec=True, side_effect=send) as send:
+            # vault_python_api.url = 'http://127.0.0.1:8200/' # v1/ is missing
+            vault_python_api.userpass_authentication('user', 'pass')
+        self.assertEqual(send.call_count, 1)
```
Comment (ewjoachim) [inline tests/test_vault_api.py]: If we're starting to write tests, I'd rather have them with pytest :)

**Comment by pilou-** [comment]
Updated:
- as suggested by yannlachiver: check the vault isn't sealed before doing anything

File: vault_cli/vault_python_api.py [hunk 2]
```diff
@@ -14,15 +15,32 @@
...
+
     d = {"password": password}
     r = s.post(url + 'auth/userpass/login/' + username, json=d, headers={})
 
     if r.status_code == requests.codes.ok:
         json_response = r.json()
         s.headers.update({'X-Vault-Token': json_response.get('auth').get('client_token')})
     else:
-        print('Wrong username or password (HTTP code: %s)' % r.status_code)
+        sys.exit('Wrong username or password (HTTP code: %s)' % r.status_code)
```
Comment (ewjoachim) [inline vault_cli/vault_python_api.py]: `click.exit()` and you can remove the sys import :)

File: vault_cli/vault_python_api.py [hunk 3]
```diff
@@ -17,12 +17,12 @@
 def userpass_authentication(username, password):
     d = {"password": password}
     r = s.post(url + 'auth/userpass/login/' + username, json=d, headers={})
-    json_response = r.json()
 
     if r.status_code == requests.codes.ok:
+        json_response = r.json()
```
Comment (ewjoachim) [inline vault_cli/vault_python_api.py]: Would `r.raise_for_status` help ?

Comment (yannlachiver) [reply vault_cli/vault_python_api.py]: I think check_status should rather be called in vault.py

Comment (yannlachiver) [reply vault_cli/vault_python_api.py]: We shoud also exit cleanly if no auth method has been specified

File: tests/test_vault_api.py (hunk 1 above)
Comment (yannlachiver) [reply tests/test_vault_api.py]: I'll not start writing tests :)

File: vault_cli/vault_python_api.py (hunk 3 above)
Comment (pilou-) [reply vault_cli/vault_python_api.py]: > I think check_status should rather be called in vault.py

Updated: `check_status` is now used with every authentication methods.

> We shoud also exit cleanly if no auth method has been specified

Program exits cleanly if no authentication method has been specified, isn't it ?

Comment (yannlachiver) [reply vault_cli/vault_python_api.py]: For now yes, but we only have authentication options, just need to keep it in mind

Comment (pilou-) [reply vault_cli/vault_python_api.py]: > Would r.raise_for_status help ?

`r.ok` might be used but Vault API [documentation](https://www.vaultproject.io/api/auth/userpass/index.html#login) mentions only HTTP code `200` even if `400` if returned when authentication fails.

File: vault_cli/vault_python_api.py (hunk 2 above)
Comment (pilou-) [reply vault_cli/vault_python_api.py]: I wasn't able to use `click.exit` could you provide more information how to use it ?

**Comment by yannlachiver** [comment]
exit(*code=0*)¶ <http://click.pocoo.org/5/api/#click.Context.exit>

Exits the application with a given exit code.


2018-04-19 16:49 GMT+02:00 Pilou <notifications@github.com>:

**Comment by yannlachiver** [comment]
We dont need the error code nor message anymore

2018-04-19 16:51 GMT+02:00 Yann Lachiver <yann.lachiver@people-doc.com>:

**Comment by yannlachiver** [comment]
si on veut etre fancy: 
```
fail(message)
Aborts the execution of the program with a specific error message.

Parameters:	message – the error message to fail with.
```

**Comment by ewjoachim** [comment]
ah oui, fail, c'est ça

**Comment by pilou-** [comment]
[click.Context.fail](https://github.com/pallets/click/blob/a87829d2c9707fb67d9a33a9f522a5d6c2b11967/click/core.py#L477) displays [usage](https://github.com/pallets/click/blob/a87829d2c9707fb67d9a33a9f522a5d6c2b11967/click/exceptions.py#L63), I am not sure this is really helpful. Instead `click.echo` utility and `click.Context.abort` could be used. What do you prefer ?

**Comment by yannlachiver** [comment]
click.exit

2018-04-19 18:00 GMT+02:00 Pilou <notifications@github.com>:

**Comment by pilou-** [comment]
Pull-request rebased and `.gitignore` updated

**Pull request merged**